pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -187,6 +199,33 @@ func (r *ReconcileHyperConverged) Reconcile(request reconcile.Request) (reconcil
 	// negative conditions (!Available, Degraded, Progressing)
 	r.conditions = nil
 
+	// Handle finalizers
+	if instance.ObjectMeta.DeletionTimestamp.IsZero() {
+		// Add the finalizer if it's not there
+		if !contains(instance.ObjectMeta.Finalizers, FinalizerName) {
+			instance.ObjectMeta.Finalizers = append(instance.ObjectMeta.Finalizers, FinalizerName)
+			// Need to requeue because finalizer update does not change metadata.generation
+			return reconcile.Result{}, r.client.Update(context.TODO(), instance)
+		}
+	} else {
+		for _, f := range []func(c client.Client, cr *hcov1alpha1.HyperConverged) error{
+			ensureCDIDeleted,
+			ensureNetworkAddonsDeleted,
+		} {
+			err = f(r.client, instance)
+			if err != nil {
+				reqLogger.Error(err, "Failed to manually delete objects")
+				return reconcile.Result{}, err
+			}
+		}
+
+		// Remove the finalizer
+                instance.ObjectMeta.Finalizers = drop(instance.ObjectMeta.Finalizers, FinalizerName)
+
```

Comment:
Please add foreground finalizer removal in case there is one, so it wont block the HCO deletion process:
```
if contains(instance.ObjectMeta.Finalizers, foregroundDeletionFinalizer) &&
    len(instance.ObjectMeta.Finalizers) == 1 {
    instance.ObjectMeta.Finalizers = drop(instance.ObjectMeta.Finalizers, foregroundDeletionFinalizer)
}
```

Reply:
are we even using the foregroundDeletionFinalizer? I don't see anywhere where we would be adding that finalizer to a HyperConverged resource.

If not, I would prefer us not adding extra logic to cleanup a finalizer we know we aren't adding.

Reply:
I searched in hco, kubevit, cnao, cdi, ssp operators repository's and there is  no usage of foreground deletion finalizer for HCO CR, so its looks like its safe not to add it.